### ISSUE TITLE
Launchpad builds doesn't have Wayland support

### ISSUE THREAD

**Issue by Sunderland93** [libretro/RetroArch#8183]
Please add `wayland-protocols` to Libretro PPA to enable Wayland support in RetroArch. Please note, that on Ubuntu 18.04 you need wayland-protocols == 1.15, otherwise you will get an error, because requeired `wayland-scanner` version is too old.

**Comment by i30817** [comment]
This is also the cause of the udev input driver no-input error (even if you added the user to the input group) and some slowness.

The cause is probably the same: fallback to a new X11 instance on Wayland. Also affects the flatpak build (but i already opened a issue for that).

**Comment by orbea** [comment]
@hizzlekizzle Since you have access the PPA builds can you do this?

**Comment by hizzlekizzle** [comment]
is this required at runtime or compile time?

**Comment by orbea** [comment]
@hizzlekizzle Compile time, its needed to generate the wayland protocols during configure.

https://github.com/libretro/RetroArch/blob/master/gfx/common/wayland/generate_wayland_protos.sh

Its probably needed at runtime too though?

**Comment by hizzlekizzle** [comment]
alright, I just added it to the compile-time dependencies in the recipe (hopefully), so it should get picked up as a runtime dependency, too, if needed

**Comment by Sunderland93** [comment]
> Its probably needed at runtime too though?

No

**Comment by Sunderland93** [comment]
@hizzlekizzle please update `wayland-protocols` package for Bionic to version 1.15

**Comment by i30817** [comment]
Why doesn't bionic update naturally to the last version available?

**Comment by Sunderland93** [comment]
Currently in Bionic's repo `wayland-protocols 1.13`. RetroArch needs >=1.15. In a build log:

```
Checking presence of package wayland-egl >= 10.1.0 ... 18.1.0
Checking presence of package wayland-cursor >= 1.12 ... 1.16.0
Checking presence of package wayland-protocols >= 1.15 ... no
Checking presence of package wayland-scanner >= 1.12 ... 1.16.0
Checking presence of package xkbcommon >= 0.3.2 ... 0.8.0
Checking presence of package xext ... 1.3.3
Checking presence of package xxf86vm ... 1.1.4
Notice: wayland libraries not found, disabling wayland support.
```

**Comment by i30817** [comment]
I see. This will make it not build in bionic until canonical makes a backport though.

**Comment by orbea** [comment]
I also noticed in configure it doesn't check if wayland-cursor is actually found for wayland support, should this be fixed?

**Comment by hizzlekizzle** [comment]
I updated the PPA recipes to hopefully include this, if anyone wants to test it.

I know appimage typically uses existing distro/ppa packages and then just bundles the libs, rather than building from source, so it will indeed just keep pulling old packages (I think trusty is the default, which is old af).

**Comment by i30817** [comment]
Yeah, fuck that. Uninstalling flatpak, building from source for now, will update back to PPA later (though i'd have liked it had a option to enable the core updater instead of it being patched out).

**Comment by hizzlekizzle** [comment]
I tried removing the patches for the online updater. Is it still doing it with the latest builds?

**Comment by orbea** [comment]
@hizzlekizzle Please remove this patch and replace it with `./configure --disable-update_assets`. Its just embarrassing how it comments out the feature while ignoring the `#ifdef`...

https://git.launchpad.net/~libretro/libretro/+git/RetroArch-debian/tree/patches/02_hide_settings.patch

**Comment by i30817** [comment]
> I tried removing the patches for the online updater. Is it still doing it with the latest builds?

I haven't been using the PPA builds for some months, i no longer know.

**Comment by Sunderland93** [comment]
@i30817 

> Semi good news is that a new Long Term Support release for Ubuntu is in April (2 months away)

In April 2020

@orbea 

> I also noticed in configure it doesn't check if wayland-cursor is actually found for wayland support, should this be fixed?

I think yes, it should be

**Comment by i30817** [comment]
Ugh, right 19.04 is not LTS (it was supposed to be but canonical decided 'nah').

**Comment by hizzlekizzle** [comment]
Alright, the online updater settings should be unpatched now (and I'm not disabling them anymore). What about the wayland stuff? Is that doing any better or do I still need to add something(s)?

**Comment by Sunderland93** [comment]
> do I still need to add something(s)?

Yes. You need to add fresh `wayland-protocols` package to Libretro PPA for Bionic and Xenial (version 1.15)

**Comment by hizzlekizzle** [comment]
I'm not sure that's going to be possible, as it won't allow me to upload anything to the PPA because we're far past the storage limit (>7 GB of files vs 2 GB limit). It doesn't seem to mind using the recipes beyond the limit but I can't just push up source packages manually.

**Comment by Sunderland93** [comment]
You may ask Launchpad maintainers to add more disk space for Libretro PPA https://answers.launchpad.net/launchpad/+addquestion

**Comment by Sunderland93** [comment]
I think it may closed by https://github.com/libretro/RetroArch/pull/8250

**Comment by Sunderland93** [comment]
Can https://github.com/libretro/RetroArch/pull/8250 be backported to RetroArch 1.7.6 (as a patch on Launchpad)?

**Comment by orbea** [comment]
Fwiw I think the idea is that `v1.7.7` will be out sooner rather than later.

**Comment by hizzlekizzle** [comment]
I went ahead and tried adding it as a patch to build for Bionic and earlier. We'll see how it goes.

**Comment by orbea** [comment]
Yea, I think there would need to be some earlier commits backported too..

**Comment by soredake** [comment]
Any progress on this?

**Comment by hizzlekizzle** [comment]
I was under the impression we should be all set here. Is it not working for you?

**Comment by soredake** [comment]
I don't see a wayland input driver list in retroarch, and i though that wayland is not enabled in builds.

**Comment by hizzlekizzle** [comment]
I don't think there is a wayland input driver. You would use udev with it, AFAIK. However, it does appear that wayland is enabled in the packaging recipe: https://git.launchpad.net/~libretro/libretro/+git/RetroArch-debian/tree/control#n38

**Comment by soredake** [comment]
It exists in code https://github.com/libretro/RetroArch/blob/b9691597d1c2996008417e905dac74a9d7b13629/input/input_driver.h#L563

**Comment by i30817** [comment]
If it exists in code, but doesn't exist in configure, it's either a misnaming in code or something that doesn't work (yet).

I don't see why you'd want a special driver other than udev for wayland because if you're on a DE using wayland its input stuff appears to take over udev - or at least i don't have the problems i have with scummvm on wayland vs on KMS login without a DE.

**Comment by gouchi** [comment]
@Sunderland93 this issue could be closed now ?

Thank you.
